lms/db/__init__.py

```diff
@@ -15,34 +15,13 @@
 __all__ = ("BASE", "init", "varchar_enum")
 
 
-class BaseClass:
-    """Functions common to all SQLAlchemy models."""
-
-    @classmethod
-    def columns(cls):
-        """Return a list of all declared SQLAlchemy column names."""
-
-        return [
-            property.key
-            for property in inspect(cls).iterate_properties
-            if isinstance(property, ColumnProperty)
-        ]
-
-    def __repr__(self):
```

Comment:
Currently, for a example a Grouping is:


```
Course(id=469, application_instance_id=113, authority_provided_id='3394ac098caa5997ed3d87023befb9f7e9ea4a38', parent_id=None, lms_id='6ed5aff92b5257ab607ec9c629844ea8586fde68', lms_name='LTI 1.3 Testing', type=<Type.COURSE: 'course'>, settings=JSONSettings({'jstor': {'enabled': False, 'site_code': None}, 'canvas': {'files_enabled': True, 'pages_enabled': False, 'groups_enabled': True, 'folders_enabled': False, 'sections_enabled': True, 'strict_section_membership': False}, 'youtube': {'enabled': True}, 'blackboard': {'files_enabled': False, 'groups_enabled': False}, 'hypothesis': {'notes': None, 'grading_comments': False, 'lms_grading_scale': False, 'auto_assigned_to_org': False, 'import_export_enabled': False, 'instructor_email_digests_enabled': True}, 'vitalsource': {'api_key': None, 'enabled': False, 'page_ranges': False, 'user_lti_param': None, 'user_lti_pattern': None, 'disable_licence_check': False}, 'desire2learn': {'client_id': None, 'files_enabled': False, 'groups_enabled': False}, 'google_drive': {'files_enabled': True}, 'microsoft_onedrive': {'files_enabled': True}}), extra={'canvas': {'custom_canvas_course_id': '319'}}, created=datetime.datetime(2023, 12, 1, 11, 35, 0, 838722), updated=datetime.datetime(2023, 12, 1, 11, 35, 0, 838722))
```


vs the default:

```<lms.models.grouping.Course at 0x7f920c2283c0>```

Reply:
Do we ever rely on the former in logs anywhere?

Reply:
I don't think we rely on this anywhere, not as far as I know.

It's helpful in development for model classes to have nice `__repr__()`'s, it can be helpful when model objects are printed in `make shell` or in `pdb` or tracebacks etc.

But I think this really needs to be a custom `__repr__()` for each model class, because which columns you want to actually include in the `__repr__()` and which are just noise depends on each model class.

The reason for this shared `__repr__()` in the base class was to avoid having to write a custom `__repr__()` in each model class because that was considered bad, but I think that was the wrong idea: separate `__repr__()`'s in model classes are actually good, you don't always want to include every single colum in a model class's `__repr__()`, and SQLAlchemy 2's dataclasses support provides a good way to autogenerate them now.